pkg/common-controller/snapshot_controller.go

```diff
@@ -425,27 +438,38 @@ func (ctrl *csiSnapshotCommonController) syncUnreadySnapshot(snapshot *crdv1.Vol
 			// update snapshot status failed
 			klog.V(4).Infof("failed to update snapshot %s status: %v", utils.SnapshotKey(snapshot), err)
 			ctrl.updateSnapshotErrorStatusWithEvent(snapshot, v1.EventTypeWarning, "SnapshotStatusUpdateFailed", fmt.Sprintf("Snapshot status update failed, %v", err))
+			ctrl.RecordSnapshotMetrics(snapshot, createSnapshotOperationName, preProvisionedSnapshotType, err)
 			return err
 		}
+
+		ctrl.RecordSnapshotMetrics(snapshot, createSnapshotOperationName, preProvisionedSnapshotType, nil)
 		return nil
 	}
+
 	// snapshot.Spec.Source.VolumeSnapshotContentName == nil - dynamically creating snapshot
+	ctrl.StartMetricsOperation(snapshot, snapshottingOperationName, dynamicSnapshotType)
```

Comment:
Do we also need to record error for snapshottingOperationName?  It looks like this only records a completion?

Reply:
I couldn't find any error cases for `snapshottingOperationName` that didn't overlap with createSnapshot. Will take another look.